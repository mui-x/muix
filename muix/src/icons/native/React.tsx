import createSvgIcon from './lib/create-svg-icon'

export default createSvgIcon(
  'react',
  'React',
  true
)