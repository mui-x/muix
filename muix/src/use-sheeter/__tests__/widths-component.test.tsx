/** @jsx platform.createElement */

import React from 'react'

import { TEngine, V } from 'reactxx-typings'
import { platform, atomizeRuleset, toClassNamesWithQuery, WIDTH, useWidths, setActWidth } from "reactxx-sheeter"

import { initPlatform, render } from "./init-platform.t"

describe("SHEETER $WIDTHS", () => {

  const doTest = (isWeb: boolean) => {
    beforeEach(() => initPlatform(isWeb))
    let ruleset

    describe("04 change width", () => {
      beforeEach(() => initPlatform(isWeb, { dataTraceFlag: 'short' }))

      const getSheetRoot = () => atomizeRuleset([
        WIDTH([0, 640], { color: 'red' }),
        WIDTH([640, 1024], { color: 'green' }),
        WIDTH(1024, { color: 'blue' }),
      ])

      const useApp = () => {
        // part of useSheeter
        const { actWidth, breakpoints, getWidthMap } = useWidths()
        const query: TEngine.WidthsQuery = {
          $widths: { actWidth, breakpoints }
        }
        const sheetRoot = React.useMemo(getSheetRoot)

        // component code
        const renderCount = React.useRef(0)
        renderCount.current++

        const root = toClassNamesWithQuery(query, sheetRoot) as any as V

        return { getWidthMap, root, renderCount: renderCount.current }
      }

      const App: React.SFC = () => {
        const { root, renderCount } = useApp()
        return <div classNames={root}>{`rendered: ${renderCount}x`}</div>
      }
      App['$c$'] = true

      const WidthsMapApp: React.SFC = () => {
        const { root, renderCount, getWidthMap } = useApp()
        const [mobile, tablet, desktop] = getWidthMap([640, 1024])
        return <div classNames={root}>
          {`${mobile ? 'mobile' : ''}${tablet ? 'tablet' : ''}${desktop ? 'desktop' : ''} (rendered: ${renderCount}x)`}
        </div>
      }
      WidthsMapApp['$c$'] = true

      const test = (App: React.ComponentType) => {
        setActWidth(640)
        const wrapper = render(<App />)
        expect(wrapper.container).toMatchSnapshot()
        setActWidth(1024)
        expect(wrapper.container).toMatchSnapshot()
        setActWidth(1025)
        expect(wrapper.container).toMatchSnapshot()
        wrapper.unmount()
      }

      it("01 use just width styles", () => test(App))
      it("02 use width flags and styles", () => test(WidthsMapApp))

    })
  }

  describe("## NATIVE ##", () => doTest(false))
  describe("## WEB ##", () => doTest(true))

})

