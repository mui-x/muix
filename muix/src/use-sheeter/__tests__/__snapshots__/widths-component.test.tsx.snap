// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SHEETER $WIDTHS ## NATIVE ## 04 change width 01 use just width styles 1`] = `
<div>
  <div
    style="color: red;"
  >
    rendered: 1x
  </div>
</div>
`;

exports[`SHEETER $WIDTHS ## NATIVE ## 04 change width 01 use just width styles 2`] = `
<div>
  <div
    style="color: green;"
  >
    rendered: 2x
  </div>
</div>
`;

exports[`SHEETER $WIDTHS ## NATIVE ## 04 change width 01 use just width styles 3`] = `
<div>
  <div
    style="color: blue;"
  >
    rendered: 3x
  </div>
</div>
`;

exports[`SHEETER $WIDTHS ## NATIVE ## 04 change width 02 use width flags and styles 1`] = `
<div>
  <div
    style="color: red;"
  >
    mobile (rendered: 1x)
  </div>
</div>
`;

exports[`SHEETER $WIDTHS ## NATIVE ## 04 change width 02 use width flags and styles 2`] = `
<div>
  <div
    style="color: green;"
  >
    tablet (rendered: 2x)
  </div>
</div>
`;

exports[`SHEETER $WIDTHS ## NATIVE ## 04 change width 02 use width flags and styles 3`] = `
<div>
  <div
    style="color: blue;"
  >
    desktop (rendered: 3x)
  </div>
</div>
`;

exports[`SHEETER $WIDTHS ## WEB ## 04 change width 01 use just width styles 1`] = `
<div>
  <div
    class="c b a"
    data-trace="
@media (min-width: 1024px).c { color:blue }
@media (min-width: 640px) and (max-width: 1023px).b { color:green }
@media (max-width: 639px).a { color:red }"
  >
    rendered: 1x
  </div>
</div>
`;

exports[`SHEETER $WIDTHS ## WEB ## 04 change width 01 use just width styles 2`] = `
<div>
  <div
    class="c b a"
    data-trace="
@media (min-width: 1024px).c { color:blue }
@media (min-width: 640px) and (max-width: 1023px).b { color:green }
@media (max-width: 639px).a { color:red }"
  >
    rendered: 1x
  </div>
</div>
`;

exports[`SHEETER $WIDTHS ## WEB ## 04 change width 01 use just width styles 3`] = `
<div>
  <div
    class="c b a"
    data-trace="
@media (min-width: 1024px).c { color:blue }
@media (min-width: 640px) and (max-width: 1023px).b { color:green }
@media (max-width: 639px).a { color:red }"
  >
    rendered: 1x
  </div>
</div>
`;

exports[`SHEETER $WIDTHS ## WEB ## 04 change width 02 use width flags and styles 1`] = `
<div>
  <div
    class="c b a"
    data-trace="
@media (min-width: 1024px).c { color:blue }
@media (min-width: 640px) and (max-width: 1023px).b { color:green }
@media (max-width: 639px).a { color:red }"
  >
    mobile (rendered: 1x)
  </div>
</div>
`;

exports[`SHEETER $WIDTHS ## WEB ## 04 change width 02 use width flags and styles 2`] = `
<div>
  <div
    class="c b a"
    data-trace="
@media (min-width: 1024px).c { color:blue }
@media (min-width: 640px) and (max-width: 1023px).b { color:green }
@media (max-width: 639px).a { color:red }"
  >
    tablet (rendered: 2x)
  </div>
</div>
`;

exports[`SHEETER $WIDTHS ## WEB ## 04 change width 02 use width flags and styles 3`] = `
<div>
  <div
    class="c b a"
    data-trace="
@media (min-width: 1024px).c { color:blue }
@media (min-width: 640px) and (max-width: 1023px).b { color:green }
@media (max-width: 639px).a { color:red }"
  >
    desktop (rendered: 3x)
  </div>
</div>
`;
